src/polyhedralGravity/model/KDTree/KDTree.cpp

```diff
@@ -30,26 +26,27 @@ namespace polyhedralGravity {
         this->getRootNode().getFaceIntersections(origin, ray, intersections);
     }
 
-
-    std::tuple<Plane, double, TriangleIndexLists<2>> KDTree::findPlane(const SplitParam &param) {// O(N^2) implementation
+    // O(N^2) implementation
+    std::tuple<Plane, double, TriangleIndexLists<2>> KDTree::findPlane(const SplitParam &param) {
         //initialize the default plane and make it costly
         double cost = std::numeric_limits<double>::infinity();
         Plane optPlane{0, param.splitDirection};
         //store the triangleSets that are implicitly generated during plane testing for later use.
         TriangleIndexLists<2> optTriangleIndexLists{};
         //each vertex proposes a split plane candidate: test for each of them, store them in buffer set to avoid duplicate testing
         std::unordered_set<double> testedPlaneCoordinates{};
-        std::for_each(param.indexBoundFaces.cbegin(), param.indexBoundFaces.cend(), [&param, &optPlane, &cost, &optTriangleIndexLists, &testedPlaneCoordinates](const size_t faceIndex) {
-            const auto &face = param.faces[faceIndex];
-            const auto [minPoint, maxPoint] = getBoundingBox(faceToVertices(face, param.vertices));
+        auto [vertex3_begin, vertex3_end] = transformIterator(param.indexBoundFaces.cbegin(), param.indexBoundFaces.cend(), param.vertices, param.faces);
+        std::for_each(vertex3_begin, vertex3_end, [&param, &optPlane, &cost, &optTriangleIndexLists, &testedPlaneCoordinates](const auto& indexAndTriplet) {
+            const auto [index, triplet] = indexAndTriplet;
+            const auto [minPoint, maxPoint] = getBoundingBox<std::array<Array3, 3>>(triplet);
```

Comment:
findPlane() now uses a transform iterator. This change has been added after the initial review.